src/lib/components/VideoCard.svelte

```diff
@@ -0,0 +1,260 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+    import { cn } from '$lib/utils/cn';
+    import Icon from './ui/icon/icon.svelte';
+
+    interface Props {
+        href: string;
+        title: string;
+        class?: string;
+    }
+
+    const { href, title, class: className = '' }: Props = $props();
+
+    let duration = $state(0);
+    let durationFormatted = $state('');
+
+    function getYoutubeId(url: string): string | null {
+        // Handle various YouTube URL formats
+        const patterns = [
+            /(?:youtube\.com\/watch\?v=|youtu\.be\/|youtube\.com\/embed\/)([^&?/]+)/,
+            /youtube\.com\/.*[?&]v=([^&]+)/
+        ];
+
+        for (const pattern of patterns) {
+            const match = url.match(pattern);
+            if (match && match[1]) {
+                return match[1];
+            }
+        }
+        return null;
+    }
+
+    function formatDuration(seconds: number): string {
+        if (!seconds) return '';
+        const mins = Math.floor(seconds / 60);
+        const secs = Math.floor(seconds % 60);
+        return `${mins}:${secs.toString().padStart(2, '0')}`;
+    }
+
+    onMount(() => {
+        const youtubeId = getYoutubeId(href);
+        if (!youtubeId) {
+            return;
+        }
+        const fetchDuration = async () => {
+            try {
+                const response = await fetch(
+                    `/api/youtube-duration?id=${encodeURIComponent(youtubeId)}`
+                );
+
+                if (!response.ok) {
+                    throw new Error(`Failed to fetch duration: ${response.status}`);
+                }
```

Comment:
lets not throw. If there's an error or the duration is not available then we can just hide the time. early return sounds better here.